--- go2.py
+++ go2.py
@@ -628,7 +628,10 @@
                 return False
 
     def __init__(self):
-        self.mainloop = GObject.MainLoop()
+        try:  # GObject.MainLoop() is deprecated, try new way
+            self.mainloop = GLib.MainLoop()
+        except AttributeError:
+            self.mainloop = GObject.MainLoop()
         self.context = self.mainloop.get_context()
         self.at_quit_func = lambda: None
         self.exception = None
